benchmark/data_frame_text_benchmark.py

```diff
@@ -0,0 +1,374 @@
+    embedding = torch.sum(
+        last_hidden_state * input_mask_expanded, dim=1) / torch.clamp(
+            input_mask_expanded.sum(1), min=1e-9)
+    return embedding.unsqueeze(1)
+
+
+def get_stype_encoder_dict(text_stype: torch_frame.stype,
+                           text_stype_encoder: StypeEncoder) -> StypeEncoder:
+    stype_encoder_dict = {
+        torch_frame.categorical:
+        EmbeddingEncoder(),
+        torch_frame.numerical:
+        LinearEncoder(),
+        # If text_stype is text_embedded,
+        # it becomes embedding after materialization
+        text_stype.parent:
+        text_stype_encoder,
+        torch_frame.multicategorical:
+        MultiCategoricalEmbeddingEncoder(),
+        torch_frame.timestamp:
+        TimestampEncoder()
+    }
+    return stype_encoder_dict
+
+
+device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+path = osp.join(osp.dirname(osp.realpath(__file__)), "..", "data")
+
+# Hyperparameters
+channels = 128
+num_layers = 4
+lr = 0.001
+epochs = 20
+num_prompts = 20
+
+if not args.finetune:
+    # text_encoder = TextToEmbedding(model=args.text_model, device=device)
+    text_encoder = HashTextEmbedder(100)
+    text_stype = torch_frame.text_embedded
+    kwargs = {
+        "text_stype":
+        text_stype,
+        "col_to_text_embedder_cfg":
+        TextEmbedderConfig(text_embedder=text_encoder, batch_size=5),
+    }
+else:
+    text_encoder = TextToEmbeddingFinetune(model=args.text_model)
+    text_stype = torch_frame.text_tokenized
+    kwargs = {
+        "text_stype":
+        text_stype,
+        "col_to_text_tokenizer_cfg":
+        TextTokenizerConfig(text_tokenizer=text_encoder.tokenize,
+                            batch_size=10000),
+    }
+
+dataset = DataFrameTextBenchmark(
+    root=path,
+    task_type=TaskType(args.task_type),
+    scale=args.scale,
+    idx=args.idx,
+    **kwargs,
+)
+
+text_model_name = args.text_model.replace('/', '')
+filename = (f"{text_model_name}_{args.task_type}_{args.scale}_"
+            f"{str(args.idx)}_{text_stype.value}_data.pt")
+dataset.materialize(path=osp.join(path, filename))
+dataset = dataset.shuffle()
+train_dataset, val_dataset, test_dataset = dataset.split()
+
+train_tensor_frame = train_dataset.tensor_frame
+val_tensor_frame = val_dataset.tensor_frame
+test_tensor_frame = test_dataset.tensor_frame
+train_loader = DataLoader(train_tensor_frame, batch_size=512, shuffle=True)
+val_loader = DataLoader(val_tensor_frame, batch_size=512)
+test_loader = DataLoader(test_tensor_frame, batch_size=512)
+
+if not args.finetune:
+    text_stype_encoder = LinearEmbeddingEncoder()
+else:
+    model_cfg = ModelConfig(model=text_encoder, out_channels=768)
+    col_to_model_cfg = {
+        col_name: model_cfg
+        for col_name in train_tensor_frame.col_names_dict[
+            torch_frame.text_tokenized]
+    }
+    text_stype_encoder = LinearModelEncoder(col_to_model_cfg=col_to_model_cfg)
```

Comment:
This should be put inside `get_stype_encoder_dict`. Otherwise, the same encoder model object will be used at every layer.

Reply:
Right, let me fix it.